native/components/message-markdown.react.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import { Component } from 'react';
+import * as React from 'react';
+import { View, ViewPropTypes } from 'react-native';
+import { merge, pick } from 'lodash';
+import PropTypes from 'prop-types';
+import SimpleMarkdown from 'simple-markdown';
+
+import styles from '../themes/markdown-styles';
+import darkStyles from '../themes/markdown-dark-styles';
+import rules from './message-markdown-rules.react';
+
+type Props = {
+  onLink: (url: string) => Promise<any>,
+  styles: typeof styles,
+  rules: any,
+  darkStyle: boolean,
+  children: React.Node,
+};
+class Markdown extends Component<Props> {
+  static propTypes = {
+    onLink: PropTypes.func,
+    styles: PropTypes.shape({
+      view: ViewPropTypes.style,
+    }),
+    rules: PropTypes.object,
+    darkStyle: PropTypes.bool,
+    children: PropTypes.node,
+  };
+
+  static defaultProps = {
+    styles: styles,
+    rules: {},
+    darkStyle: false,
+  };
+  parse: any;
+  renderer: any;
+
+  constructor(props: Props) {
+    super(props);
+    const opts = {
+      onLink: props.onLink,
+      rules: props.rules,
+    };
+
+    const style = props.darkStyle ? darkStyles : styles;
+    var customRules = rules(style, opts);
+
+    var customMarkdownRules = pick(SimpleMarkdown.defaultRules, [
```

Comment:
Why do we need these?

Reply:
Maybe the name is not the best, but basically we need to choose which rules from `SimpleMarkdown.defaultRules` we want to use, if we just use all of them and user types something recognized by `simple-markdown` we'll get an error because we don't have components to match and create in our rules

Reply:
I guess I'm confused why we need the `defaultRules` when we've defined the rules in `message-markdown-rules.react.js` already

Reply:
`defaultRules` contain some additional data, like order, that our rules don't

Reply:
Make sense. In that case, let's define those extra data inline in `message-markdown-rules.react.js`:

```js
{
  autolink: {
    ...SimpleMarkdown.defaultRules.autolink,
    react: function(
      node: SingleASTNode,
      output: Output<ReactElement>,
      state: State,
    ) {
      [...]
    },
  },
}
```